### ISSUE TITLE
Auto-complete constructor init list should use {}

### ISSUE THREAD

**Issue by JVApen** [clangd/clangd#2214]
`````
struct S
{
   int xyz;
   S() : x^ {}
}
````
Auto-complete at ^ suggests to insert `xyz($0)` while `xyz{$0}` would prevent narrowing conversions.
C++ core guidelines suggest to use {}, see https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-list

This is easily adapted in `Sema::CodeCompleteConstructorInitializer()` > `AddDefaultCtorInit` by using CK_LeftBrace instead of CK_LeftParen (and same for Right brace).

@HighCommander4: As this only works from C++11 on, does this need some specific check on language version? Does this somehow need some configuration to choose?

**Comment by HighCommander4** [comment]
> As this only works from C++11 on, does this need some specific check on language version?

Yes, see [this](https://searchfox.org/llvm/rev/e8c07f7458285c6fb2eddff5b7914519de10474d/clang/lib/Sema/SemaCodeComplete.cpp#1828) for an example.

> Does this somehow need some configuration to choose?

Good question. I guess existing projects are likely to have a convention about this and clangd is more helpful if it can be made to follow the convention.

So, yeah, a config option for this would probably be helpful. It could be added under [Completion](https://clangd.llvm.org/config.html#completion). This is a bit annoying to do because to get an option from clangd to Sema code we have to plumb it through `CodeCompleteOptions`, but it's doable.

**Comment by mauriba** [comment]
+1, experiencing the same issue. Is there any quickfix to let clangd autocomplete curly brackets instead of rounded ones for member initialization? Would be very appreciated.

**Comment by HighCommander4** [comment]
There is not currently.

Contributions to implement the configuration option described above are welcome.
